pkg/hs/urbit-king/lib/Urbit/King/Main.hs

```diff
@@ -519,45 +525,48 @@ newShip CLI.New{..} opts = do
       Just x  -> x
       Nothing -> "./" <> unpack name
 
-    nameFromShip :: Ship -> RIO KingEnv Text
+    nameFromShip :: Ship -> RIO RunningEnv Text
     nameFromShip s = name
       where
         nameWithSig = Ob.renderPatp $ Ob.patp $ fromIntegral s
         name = case stripPrefix "~" nameWithSig of
           Nothing -> error "Urbit.ob didn't produce string with ~"
           Just x  -> pure x
 
-    bootFromSeed :: MultiEyreApi -> Pill -> Seed -> RIO KingEnv ()
-    bootFromSeed multi pill seed = do
+    bootFromSeed :: Pill -> Seed -> RIO RunningEnv ()
+    bootFromSeed pill seed = do
       ethReturn <- dawnVent seed
 
       case ethReturn of
         Left x -> error $ unpack x
         Right dawn -> do
           let ship = sShip $ dSeed dawn
           name <- nameFromShip ship
-          runTryBootFromPill multi pill name ship (Dawn dawn)
+          runTryBootFromPill pill name ship (Dawn dawn)
 
     -- Now that we have all the information for running an application with a
     -- PierConfig, do so.
-    runTryBootFromPill multi pill name ship bootEvent = do
-      vKill <- view kingEnvKillSignal
+    runTryBootFromPill :: Pill -> Text -> Ship -> LegacyBootEvent
+                       -> RIO RunningEnv ()
+    runTryBootFromPill pill name ship bootEvent = do
+      env <- ask
+      let vKill = (env ^. kingEnvL) ^. kingEnvKillSignal
       let pierConfig = toPierConfig (pierPath name) opts
       let networkConfig = toNetworkConfig opts
       runPierEnv pierConfig networkConfig vKill $
-        tryBootFromPill True pill nLite ship bootEvent multi
+        tryBootFromPill True pill nLite ship bootEvent
 ------  tryBootFromPill (CLI.oExit opts) pill nLite flags ship bootEvent
 
-runShipEnv :: CLI.Run -> CLI.Opts -> TMVar () -> RIO PierEnv a -> RIO KingEnv a
+runShipEnv :: CLI.Run -> CLI.Opts -> TMVar () -> RIO PierEnv a -> RIO RunningEnv a
```

Comment:
No, because runPierEnv requires a RunningEnv, since a RunningEnv. Fixing this is a longer term issue because of the dependency stack issue talked about above.